utils/logalloc.cc

```diff
@@ -748,9 +748,13 @@ class segment_store {
     size_t max_segments() const noexcept {
         return _std_memory_available / segment::size;
     }
-    bool can_allocate_more_segments() noexcept {
-        auto i = find_empty();
-        return i != _segments.end();
+    bool can_allocate_more_segments() const noexcept {
+        for (auto it = _segments.cbegin() + 1; it != _segments.cend(); ++it) {
```

Comment:
`find_empty` is non-const.
Instead of open-coding the loop I can use:
```
        auto i = std::find(_segments.cbegin() + 1, _segments.cend(), nullptr);
        return i != _segments.cend();
```

not sure if it is any better, but it reflects a smaller change.

Reply:
> Can't this be just fixed?

We can add a const overload for find_empty, yes.